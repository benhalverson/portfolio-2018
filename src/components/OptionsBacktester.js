import React from 'react';
import IGOTdefaultview from '../images/IGOT-default-view.png';
import IGOTadvanceddefaultview from '../images/IGOT-default-view.png';
import IGOTtableexpanded from '../images/IGOT-table-expanded.png';
import IGOTtablechart from '../images/IGOT-table-chart.png';
import IGOTsearchresults from '../images/IGOT-search-results.png';
import IGOTdatepicker from '../images/IGOT-datepicker.png';

const OptionsBacktester = () => (
  <div>
    <div className="title">E-Trade Options Backtester</div>
    <div className="body">
      <p className="description">
        I was the lead on this new product. I built the first prototype in
        Angular.js and version 1 of this product.
      </p>
      <div className="screenshots">
        <img className="col" src={IGOTdefaultview} alt="" />
        <img className="col" src={IGOTadvanceddefaultview} alt="" />
        <img className="col" src={IGOTtableexpanded} alt="" />
        <img className="col" src={IGOTtablechart} alt="" />
        <img className="col" src={IGOTsearchresults} alt="" />
        <img className="col" src={IGOTdatepicker} alt="" />
      </div>
    </div>
  </div>
);

export default OptionsBacktester;
