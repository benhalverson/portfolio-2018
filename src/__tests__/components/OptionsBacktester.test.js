import React from 'react';
import { shallow } from 'enzyme';

import OptionsBacktester from '../../components/OptionsBacktester';
test('it should render the OptionsBacktester component', () => {
  const wrapper = shallow(<OptionsBacktester />);
  expect(wrapper).toMatchSnapshot();
});
