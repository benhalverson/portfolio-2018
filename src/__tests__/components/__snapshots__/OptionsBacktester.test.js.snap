// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the OptionsBacktester component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsBacktester />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
>
          E-Trade Options Backtester
</div>,
        <div
          className="body"
>
          <p
                    className="description"
          >
                    I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
          </p>
          <div
                    className="screenshots"
          >
                    <img
                              alt=""
                              className="col"
                              src="IGOT-default-view.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="IGOT-default-view.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="IGOT-table-expanded.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="IGOT-table-chart.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="IGOT-search-results.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="IGOT-datepicker.png"
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-Trade Options Backtester",
          "className": "title",
        },
        "ref": null,
        "rendered": "E-Trade Options Backtester",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description"
>
              I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
</p>,
            <div
              className="screenshots"
>
              <img
                            alt=""
                            className="col"
                            src="IGOT-default-view.png"
              />
              <img
                            alt=""
                            className="col"
                            src="IGOT-default-view.png"
              />
              <img
                            alt=""
                            className="col"
                            src="IGOT-table-expanded.png"
              />
              <img
                            alt=""
                            className="col"
                            src="IGOT-table-chart.png"
              />
              <img
                            alt=""
                            className="col"
                            src="IGOT-search-results.png"
              />
              <img
                            alt=""
                            className="col"
                            src="IGOT-datepicker.png"
              />
</div>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
              "className": "description",
            },
            "ref": null,
            "rendered": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="col"
                  src="IGOT-default-view.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="IGOT-default-view.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="IGOT-table-expanded.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="IGOT-table-chart.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="IGOT-search-results.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="IGOT-datepicker.png"
/>,
              ],
              "className": "screenshots",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-default-view.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-default-view.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-table-expanded.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-table-chart.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-search-results.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "IGOT-datepicker.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
>
            E-Trade Options Backtester
</div>,
          <div
            className="body"
>
            <p
                        className="description"
            >
                        I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
            </p>
            <div
                        className="screenshots"
            >
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-default-view.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-default-view.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-table-expanded.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-table-chart.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-search-results.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="IGOT-datepicker.png"
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-Trade Options Backtester",
            "className": "title",
          },
          "ref": null,
          "rendered": "E-Trade Options Backtester",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description"
>
                I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
</p>,
              <div
                className="screenshots"
>
                <img
                                alt=""
                                className="col"
                                src="IGOT-default-view.png"
                />
                <img
                                alt=""
                                className="col"
                                src="IGOT-default-view.png"
                />
                <img
                                alt=""
                                className="col"
                                src="IGOT-table-expanded.png"
                />
                <img
                                alt=""
                                className="col"
                                src="IGOT-table-chart.png"
                />
                <img
                                alt=""
                                className="col"
                                src="IGOT-search-results.png"
                />
                <img
                                alt=""
                                className="col"
                                src="IGOT-datepicker.png"
                />
</div>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
                "className": "description",
              },
              "ref": null,
              "rendered": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="col"
                    src="IGOT-default-view.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="IGOT-default-view.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="IGOT-table-expanded.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="IGOT-table-chart.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="IGOT-search-results.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="IGOT-datepicker.png"
/>,
                ],
                "className": "screenshots",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-default-view.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-default-view.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-table-expanded.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-table-chart.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-search-results.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "IGOT-datepicker.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
